haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -240,28 +245,73 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
             deserialize_secrets_inplace(data["init_parameters"]["generator_api_params"], keys=["api_key"])
         return default_from_dict(cls, data)
 
+    def _extract_metadata_and_update_doc(self, document, errors, content):
+        """
+        Extract metadata from the content and update the document with the extracted metadata. If the extraction fails,
+        the error message will be stored in `llm_errors`.
+
+        :param document: Document to be updated with the extracted metadata.
+        :param errors: Dictionary to store error messages if the extraction fails.
+        :param content: Content to extract metadata from.
+        """
+        prompt_with_doc = self.builder.run(
+            template=self.prompt,
+            template_variables={self.prompt_variable:content}
+        )
+        print(f"Prompt with doc: {prompt_with_doc}")
+        result = self.llm_provider.run(prompt=prompt_with_doc["prompt"])
+        llm_answer = result["replies"][0]
+        if self.is_valid_json_and_has_expected_keys(expected=self.expected_keys, received=llm_answer):
+            extracted_metadata = json.loads(llm_answer)
+            for k in self.expected_keys:
+                document.meta[k] = extracted_metadata[k]
+        else:
+            errors[document.id] = llm_answer
+
 
     @component.output_types(documents=List[Document], errors=Dict[str, Any])
-    def run(self, documents: List[Document]) -> Dict[str, Any]:
+    def run(self, documents: List[Document], page_range: Optional[List[str]] = None):    # pylint: disable=R0914
```

Comment:
Could we also add `page_range` as an init parameter? For me I'd probably only use it there and not so much in the run method. So I think it would be good to have in both.